### PR TITLE
Update ubi micro image

### PR DESCRIPTION
**Pull request by EmilyKatdell**
# Description
Update ubi micro image

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| [790](https://github.com/dell/csm/issues/790)|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Manual testing done

### CODE REVIEW

File: Dockerfiles/Dockerfile.init [hunk 1]
```diff
@@ -3,15 +3,17 @@
...
+
 FROM $BASEIMAGE AS init
 WORKDIR /upgrade
 COPY scripts/upgrade ./
-RUN microdnf update -y && \
-    microdnf install -y shadow-utils wget && \
-    wget -v https://dl.k8s.io/release/v1.25.7/bin/linux/amd64/kubectl && \
-    microdnf -y remove shadow-utils wget && \
-    microdnf clean all && \
+RUN mkdir -p ./kubectl /usr/local/bin && \
```
Comment (donatwork) [inline Dockerfiles/Dockerfile.init:15]: This line means to create two directories. Don't understand why you are creating a directory ./kubectl.

File: Dockerfiles/Dockerfile.init [hunk 2]
```diff
@@ -3,15 +3,17 @@
 # some arguments that must be supplied
 ARG BASEIMAGE
 
+FROM amaas-eos-mw1.cec.lab.emc.com:5106/dcm/go-builder-image:latest as builder
+WORKDIR /upgrade
+COPY scripts/upgrade ./
```
[the remark this answers is not in the document]
Comment (donatwork) [reply Dockerfiles/Dockerfile.init:8]: @EmilyKatdell The new changes are copying the scripts to the builder image and not the init image. All you need to do in the builder image is to wget the kubecte file. Then in the init image you copy the kubectl (--from=builder) and the migrate_rg.sh script. Also set the execute permissions there.

[the remark this answers is not in the document]
Comment (donatwork) [reply Dockerfiles/Dockerfile.init]: @EmilyKatdell I updated the Docerfile.init. Can you please test the new image.

File: Dockerfiles/Dockerfile.init [hunk 3]
```diff
@@ -3,15 +3,15 @@
 # some arguments that must be supplied
 ARG BASEIMAGE
 
+FROM golang:1.20 as builder
```
Comment (santhoshatdell) [inline Dockerfiles/Dockerfile.init:6]: AS

File: image.mk [hunk 4]
```diff
@@ -4,13 +4,13 @@ include semver.mk
 ifneq (, $(shell which podman 2>/dev/null))
         CONTAINER_TOOL ?= podman
 else
-        CONTAINER_TOOL ?= docker
+        CONTAINER_TOOL ?= docker
```
Comment (santhoshatdell) [inline image.mk:7]: Additional space can be removed.

**Review by santhoshatdell — APPROVED** [review]

File: Dockerfiles/Dockerfile.init (hunk 3 above)
Comment (donatwork) [reply Dockerfiles/Dockerfile.init:6]: The Dockerfile has this in lowercase as well. We can change in the future.

File: image.mk (hunk 4 above)
Comment (donatwork) [reply image.mk:7]: We can fix this later. We need to get everyone to set their IDEs to trim trailing whitespace.

**Review by harshaatdell — APPROVED** [review]
Nice team work.

**Pull request merged**